python/ray/air/checkpoint.py

```diff
@@ -331,6 +342,17 @@ def _to_directory(self, path: str) -> None:
             # This is a object ref or dict
             data_dict = self.to_dict()
             if _FS_CHECKPOINT_KEY in data_dict:
+                # On top of that, there may be some dict data.
```

Comment:
Instead of this checkpoint vs metadata stuff, why don't we store each (k, v) pair as a separate file in the directory?